src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -47,25 +62,15 @@ public function create(DocumentGroupDefinitionInterface $definition)
             throw new InvalidArgumentException('Definition must be an instance of ' . DocumentGroupDefinition::class);
         }
 
-        // TODO: move this to a custom factory so that we can inject the factory in this class and then have the
-        // middleware as dependency of that factory.
+        $strategies = $this->stategies;
+        $strategies[] = new File(
```

Comment:
I would prefer that the File factory doesn't instantiate the necessary adapters at constructor time but when a `get` or `create` method is invoked (which then accepts a FlySystemAdapter or destination uri). That way we can move the instantiation of these objects out of  this class into the DIC.